app/controllers/responsiblepeople/RemoveResponsiblePersonController.scala

```diff
@@ -49,6 +49,8 @@ class RemoveResponsiblePersonController @Inject () (
         rp <- getData[ResponsiblePerson](index)
         status <- statusService.getStatus
       } yield rp match {
+        case Some(person) if (person.lineId.isDefined && !person.isComplete) =>
+          Redirect(routes.WhatYouNeedController.get(index, flow))
```

Comment:
There's one matter I need to clarify... this will redirect to WhatYouNeedController every user with some incomplete data and line id. So for example people with missing fit and proper answer when renewing supervision. Is this desired behaviour? Example application: XDML00000567893

Reply:
As all RPs with a line ID need to pass schema validation I think its acceptable to push them back through the flow prior to deletion.

Reply:
Ok, thanks :) it's clear now